src/pages/api/access_codes/create.ts

```diff
@@ -11,21 +11,21 @@ const json_body = z
     code: z.string().optional(),
     starts_at: timestamp.optional(),
     ends_at: timestamp.optional(),
-    use_backup_access_code_pool: z.boolean().optional(),
+    use_backup_access_code_pool: z.boolean().optional().default(false),
   })
   .refine((value) => {
     if (
-      (value.starts_at && !value.ends_at) ||
-      (value.ends_at && !value.starts_at)
+      (value.starts_at != null && value.ends_at == null) ||
+      (value.ends_at != null && value.starts_at == null)
     ) {
       return false
     }
     return true
   }, "Both starts_at and ends_at must be provided if one is")
   .refine((value) => {
     if (
-      !(value.starts_at || value.ends_at) &&
-      value.use_backup_access_code_pool
+      value.use_backup_access_code_pool &&
+      (value.starts_at == null || value.ends_at == null)
```

Comment:
Can someone double check this logic is right? If either of these is null then it's not a timebound code right?

Reply:
So the identical conditional would be `value.starts_at == null && value.ends_at == null` but if one of them is defined then it should be time bound too right?

Reply:
looks like the logic on connect is `!(value.starts_at || value.ends_at)`